tcl/envmngt.tcl.in

```diff
@@ -1815,17 +1815,32 @@ proc add-path {cmd mode dflbhv args} {
 
    set val [get-env $var]
 
-   foreach dir $path_list {
-      if {![info exists countarr($dir)] || $allow_dup} {
-         # ignore env var set empty if no empty entry found in reference
-         # counter array (sometimes var is cleared by setting it empty not
-         # unsetting it)
-         if {$val ne {} || [info exists countarr()]} {
-            set sep [expr {$val eq $separator ? {} : $separator}]
-            set val [expr {$bhv eq {prepend} ? "$dir$sep$val" :\
-               "$val$sep$dir"}]
+   if {[getConf lmod_path_rule]} {
+      set mpath_list [split $val $separator]
+      foreach dir $path_list {
+         if {$bhv eq {prepend}} {
+            set mpath_list "$dir [lsearch -inline -all -not -exact\
```

Comment:
I recommend having the path removal command separated, as it is the same whether we append or we prepend.